collection.js

```diff
@@ -1,63 +1,60 @@
 var EventEmitter = require('events').EventEmitter;
+var url = require('url');
+var util = require('util');
+var rels = require('zetta-rels');
+var uuid = require('node-uuid');
 
 var StatsCollector = module.exports = function() {
-  this.emitter = new EventEmitter();
-  this.server = null;
-};
-
-StatsCollector.prototype.on = function() {
-  this.emitter.on.apply(this.emitter, arguments);
+  EventEmitter.call(this);
 };
+util.inherits(StatsCollector, EventEmitter);
 
 StatsCollector.prototype.collect = function() {
   return this._collect.bind(this);
 };
-StatsCollector.prototype._collect = function(server) {
+
+StatsCollector.prototype._collect = function(runtime) {
   var self = this;
-  this.server = server;
 
-  var servers = []; // keep a list of peers subscribed to, to filter duplicates
-  server.pubsub.subscribe('_peer/connect', function(e, msg) {
-    if (servers.indexOf(msg.peer.name) > -1) {
+  runtime.pubsub.subscribe('_peer/connect', function(ev, msg) {
+    var topic = 'query:' + uuid.v4() + '/where type is not missing';
```

Comment:
Good catch.  I was initially using `'query:' + msg.peer.name`, and for some reason, that wasn't working well.  I looked at the remote device code, pulled this piece in, and it worked just fine.  But then I didn't account for that anywhere else.  I'll revisit this.  Good catch.